### PR TITLE
Issue 946 | Optimistic lock

### LINKED ISSUE #946: job.result sometimes returns an empty dict when using save_result
The following remote program sometimes returns an empty dictionary as the result

```
# program.py
from quantum_serverless import save_result

my_list = [1, 2, 3, 4]
save_result({"results": my_list})
```

<Job | b90b6513-4a93-451b-871f-39681b26cbb7> returned an empty dictionary

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Optimistic lock implementation based on https://github.com/Qiskit-Extensions/quantum-serverless/pull/956

### Details and comments

Looks like https://github.com/Qiskit-Extensions/quantum-serverless/pull/959 did not fixed an issue.
It looks like `update_fields` only changes query, but commit is still happening at the same time, therefore causing error. 

Closes #946 

Additionally  removed ray client logs call from `/logs` view, since logs are updated in management commands.

### CODE REVIEW

File: gateway/api/utils.py [hunk 1]
```diff
@@ -165,3 +167,36 @@ def decrypt_env_vars(env_vars: Dict[str, str]) -> Dict[str, str]:
...
+            else:
+                model.save()
+            saved = True
+        except RecordModifiedError:
+            logger.warning(
+                "Model[%s] record has not been updated due to lock. Retrying. Attempts left %s",
+                model,
+                attempts_left,
+            )
+            continue
```
Comment (akihikokuroda) [inline gateway/api/utils.py:200]: I believe that the model record must be refetched and updates re-applied to the fields to resolve `RecordModifiedError` exception.

Comment (IceKhan13) [reply gateway/api/utils.py:200]: aaaah, good to know!

Comment (akihikokuroda) [reply gateway/api/utils.py:200]: I think that the record need to be refetched in this loop, too.   

My understanding of the this concurrency is that it check if the version of the record in the db is same as the one to be saved.  The version is updated every time the record is saved.  If the version don't match, somebody has updated the the record while the recode is fetched and being saved.

Comment (akihikokuroda) [reply gateway/api/utils.py:200]: I haven't verified but I was wondering if the `version` field need to be added for the `save()` call to make the `concurrency` work.

Comment (IceKhan13) [reply gateway/api/utils.py:200]: I think I will remove this loop

Comment (IceKhan13) [reply gateway/api/utils.py:200]: > I haven't verified but I was wondering if the version field need to be added for the save() call to make the concurrency work.

and remove update fields too, I think

Comment (IceKhan13) [reply gateway/api/utils.py]: updated

**Review by akihikokuroda — APPROVED** [review]
LGTM  Thanks!

**Pull request merged**